aws_lambda_builders/workflows/java_maven/maven.py

```diff
@@ -0,0 +1,52 @@
+"""
+Wrapper around calls to Maven through a subprocess.
+"""
+
+import logging
+import subprocess
+
+LOG = logging.getLogger(__name__)
+
+
+class MavenExecutionError(Exception):
+    MESSAGE = "Maven Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+
+class SubprocessMaven(object):
+
+    def __init__(self, maven_binary, os_utils=None):
+        if maven_binary is None:
+            raise ValueError("Must provide Maven BinaryPath")
+        self.maven_binary = maven_binary
+        if os_utils is None:
+            raise ValueError("Must provide OSUtils")
+        self.os_utils = os_utils
+
+    def retrieve_module_name(self, scratch_dir):
+        args = ['-q', '-Dexec.executable=echo', '-Dexec.args=${project.artifactId}',
+                'exec:exec', '--non-recursive']
+        ret_code, stdout, stderr = self._run(args, scratch_dir)
+        if ret_code != 0:
+            raise MavenExecutionError(message=stderr.decode('utf8').strip())
+        return stdout.decode('utf8').strip()
+
+    def build(self, scratch_dir, module_name):
```

Comment:
Might be a good idea to `LOG.info(stdout) or LOG.debug(stdout)`. When the build fails, it could be useful to inspect maven output to see what happened.